### PR TITLE
Drop tool migration scripts and related code

### PR DESCRIPTION
**Pull request by jdavcs**
This is a follow-up to a discussion we had at a backend WG meeting.
The functionality is no longer used and, due to its reliance on SQLAlchemy Migrate, is a blocker for #12541 (SQLAlchemy 2.0).

I've left the `migrated_tools_config` config option, but marked it as deprecated and changed its description accordingly. That way, we still check for that file, in case there's a deployment that still has one. I've removed the `check_migrate_tools` config option and (hopefully) all related functionality.

EDIT: I'm not sure to what extent this might affect #636 (thanks for the reference, @nsoranzo!)

## How to test the changes?
(Select all options that apply)
- [ ] I've included appropriate [automated tests](https://docs.galaxyproject.org/en/latest/dev/writing_tests.html).
- [x] This is a refactoring of components with existing test coverage.
- [ ] Instructions for manual testing are as follows:
  1. [add testing steps and prerequisites here if you didn't write automated tests covering all your changes]

## License
- [x] I agree to license these contributions under [Galaxy's current license](https://github.com/galaxyproject/galaxy/blob/dev/LICENSE.txt).
- [x] I agree to allow the Galaxy committers to license these and all my past contributions to the core galaxy codebase under the [MIT license](https://opensource.org/licenses/MIT). If this condition is an issue, uncheck and just let us know why with an e-mail to galaxy-committers@lists.galaxyproject.org.

### CODE REVIEW

**Comment by ic4f** [comment]
All the red is due to a disruption of service on circleci's end. I'll rerun the tests when that's fixed.

File: test/unit/config/config_manage/1607_root_filters/config/galaxy.ini [hunk 1]
```diff
@@ -144,15 +144,6 @@ paste.app_factory = galaxy.web.buildapp:app_factory
...
 
-# Enable / disable checking if any tools defined in the above non-shed
-# tool_config_files (i.e., tool_conf.xml) have been migrated from the Galaxy
-# code distribution to the Tool Shed.  This setting should generally be set to
-# False only for development Galaxy environments that are often rebuilt from
-# scratch where migrated tools do not need to be available in the Galaxy tool
-# panel.  If the following setting remains commented, the default setting will
-# be True.
-#check_migrate_tools = True
-
```
Comment (nsoranzo) [inline test/unit/config/config_manage/1607_root_filters/config/galaxy.ini]: I don't think this and the following file should be modified?

Comment (jdavcs) [reply test/unit/config/config_manage/1607_root_filters/config/galaxy.ini]: I don't mind. But why would we need this option to be part of a test file if it's removed? (I don't mind either way, just double-checking).

Comment (mvdbeek) [reply test/unit/config/config_manage/1607_root_filters/config/galaxy.ini]: I guess the point is that we have a config as it used to be, so we can test the config_manage script. It does make sense to keep it imo.

Comment (jdavcs) [reply test/unit/config/config_manage/1607_root_filters/config/galaxy.ini]: Done!

**Comment by jmchilton** [comment]
```+65 −2,450``` 😺 🏆 .

Looks wonderful to me!

**Review by jmchilton — APPROVED** [review]

**Pull request merged**